libsrc/common/getcpu.s

```diff
@@ -26,52 +27,119 @@
 .p816                           ; Enable 65816 instructions
 
 _getcpu:
+
         lda     #0
         inc     a               ; .byte $1A ; nop on nmos, inc on every cmos
         cmp     #1
-        bcc     @L8
+        bcc     @IsNMOS
 
 ; This is at least a 65C02, check for a 65CE02/4510
 
         .byte   $42,$EA         ; neg on 65CE02/4510, nop #$EA on 65C02, wdm $EA on 65816
         cmp     #1
-        beq     @L6
+        beq     @HasINCA
 
 ; This is at least a 65CE02, check for 4510
 
         lda     #5              ; CPU_65CE02 constant
+        ldx     #0              ; to make sure MAP doesn't do anything, the upper nybl of X and Z must be clear
         .byte   $5C             ; map on 4510, aug on 65CE02 (acts like 4 byte nop)
         lda     #3              ; CPU_4510 constant
         nop
-        bne     @L9
+        cmp     #5
+        beq     @LoadXAndReturn
+
+; It is either a 4510 (C65) or a 45GS02 (MEGA65)
+
+        ; 45GS02 supports 32-bit ZP indirect, so use that to check CPU type
+        ; without requiring a functioning MEGA65 hypervisor.
+        ; We setup a read of $200FF, then store a different value in $00FF
+        ; and then re-read $200FF to see if it is unchanged.
+
+        ; Save the 32-bit ZP pointer and data byte
+        ldx     #4
+@L10:   lda     $fb,x
+        sta     @GetCPUTemp,x
+        dex
+        bpl     @L10
+
+        ; Setup 32-bit pointer to $000200FF
+        lda     #$ff
+        sta     $fb
+        lda     #$00
+        sta     $fc
+        sta     $fe
+        lda     #$02
```

Comment:
Rather #2 ?

Reply:
Here I used #$02 to indicate that it is part of the 32-bit address indicated in the comment on line 66, which would not be as obvious if #0 were used instead. But happy to go with your preference.

Reply:
Actually, for these, would it be better to use LDA #<$0002 / LDA #>$0002 , LDA #<$00FF / LDA #$>00FF to make it clearer to the reader that the constants are parts of addresses?

Reply:
> Actually, for these, would it be better to use `LDA #<$0002` / `LDA #>$0002` , `LDA #<$00FF` / `LDA #>$00FF` to make it clearer to the reader that the constants are parts of addresses?

Definitely yes!  And, don't forget about the `^` operator which can pull the third (bank) byte out of a 24-bit number.

Reply:
Full ACK. I just didn't notice that these is about addr bytes.

Reply:
Now uses <, >, and ^ to make it all more obvious.